### ISSUE TITLE
Server connection can't handle CNAME records?

### ISSUE THREAD

**Issue by whilenot-dev** [jellyfin/jellyfin-androidtv#4053]
### Describe the bug

**tl;dr** The _Jellyfin_ Android TV client can handle IPs or A records just fine, but can't handle CNAME records as server connections.

## Description

I'm running _Jellyfin_ behind a _[Caddy](https://caddyserver.com/)_ reverse proxy and use _[pihole](https://pi-hole.net/)_ as local DNS proxy. I want to connect to the _Jellyfin_ server via a custom DNS record within the _Jellyfin_ Android TV client. I just noticed that the client can connect to my server via IPs or A records just fine, but not via CNAME records.

That was just a minor annoyance and might not be worth a fix, but at least it should be documented somewhere if it isn't my fault and the intended behavior.

## Workaround

Don't use CNAME records, just use A records or IPs to connect to a _Jellyfin_ server.

### Versions

- _Jellyfin_ (Server): `10.9.11`
- _Jellyfin_ Android TV (Client): `0.17.6`
- Android: `Base on Android 7.1` (acc. to [VAVA 4K Laser TV](https://www.vava.com/products/4k-ultra-short-throw-laser-projector/))

#### Summary

_Jellyfin_ (Server) runs on a NAS (`machine-a.local`) and _Caddy_ and _pihole_ run on a Raspberry Pi (`machine-b.local`). Everything runs in the same network (domain name `.local`). My Android TV box is configured to use _pihole_ as DNS server.

#### In Detail

- `HOSTNAME_MACHINE_A` (`machine-a.local`) runs the _Jellyfin_ server on port `8096`
- `HOSTNAME_MACHINE_B` (`machine-b.local`) runs:
  - the _Caddy_ reverse proxy on ports `80`, `443`, `8096`
  - the `pihole` DNS proxy on port `53`

##### _Jellyfin_ server

Run the _Jellyfin_ server on a machine (p.e. `${HOSTNAME_MACHINE_A}`):

```
# compose.yml

services:
  jellyfin:
    container_name: jellyfin
    image: jellyfin/jellyfin:10.9.11
    ports:
      - 8096:8096
    environment:
      - JELLYFIN_PublishedServerUrl=https://jellyfin.service
    restart: unless-stopped
    volumes:
      - ./cache:/cache
      - ./config:/config
```

##### _Caddy_ reverse proxy

Run the _Caddy_ reverse proxy on another machine (p.e. `${HOSTNAME_MACHINE_B}`):

```
# compose.yml

services:
  caddy:
    container_name: caddy
    image: caddy:2.8.4-alpine
    ports:
      - "80:80"
      - "443:443/tcp"
      - "443:443/udp"
      - "8096:8096"
    networks:
      - caddy
    restart: unless-stopped
    volumes:
      - "./Caddyfile:/etc/caddy/Caddyfile:ro"
      - config:/config
      - data:/data

networks:
  caddy:

volumes:
  config:
  data:
```

Provide a reverse proxy config for _Jellyfin_ (and _pihole_):

```
# Caddyfile

# jellyfin HTTP
http://jellyfin.service:8096 {
        reverse_proxy ${HOSTNAME_MACHINE_A}:8096
}

# jellyfin HTTPS
jellyfin.service {
        reverse_proxy ${HOSTNAME_MACHINE_A}:8096
        tls internal
}

pi.hole {
        reverse_proxy pihole:80
        tls internal
}
```

##### _pihole_ DNS proxy

Run the _pihole_ DNS proxy on another machine (p.e. also on `${HOSTNAME_MACHINE_B}`):

```
services:
  pihole:
    container_name: pihole
    image: pihole/pihole:2024.07.0
    ports:
      - "53:53/tcp"
      - "53:53/udp"
    networks:
      - caddy
    volumes:
      - './etc/dnsmasq.d:/etc/dnsmasq.d'
      - './etc/pihole:/etc/pihole'
      - './var/log/pihole.log:/var/log/pihole.log'
    restart: unless-stopped

networks:
  caddy:
    external: true
```

#### Prerequisites

Make sure _Jellyfin_ runs and is accessible via web client on `http://${HOSTNAME_MACHINE_A}:8096` (eg. `http://machine-a.local:8096`) and `http://${IPV4_MACHINE_A}:8096` (eg. `http://10.0.0.1:8096`).

Creating a server connection in the Android TV client will work via `http://${HOSTNAME_MACHINE_A}:8096` or `http://${IPV4_MACHINE_A}:8096`.

#### Create a CNAME record

Setup a local CNAME record to the _Caddy_ reverse proxy in the _pihole_ admin page:

```
jellyfin.service    ${HOSTNAME_MACHINE_B}
...eg.
jellyfin.service    machine-b.local
```

Then make sure you can connect to the _Jellyfin_ web client via DNS records: `https://jellyfin.service`  (HTTPS with self-signed TLS certificate) or `http://jellyfin.service:8096` (HTTP).

Creating a server connection now in the Android TV client won't work via `http://jellyfin.service:8096` (`https://jellyfin.service` wouldn't work anyway in my case, because the TLS certificate is self-signed).

`http://${HOSTNAME_MACHINE_A}:8096`, and `http://${IPV4_MACHINE_A}:8096` will still work just as before.

#### Change to an A record

Remove the CNAME record and instead create an A record in the _pihole_ admin page:

```
jellyfin.service    ${IPV4_MACHINE_B}
...eg.
jellyfin.service    10.0.0.2
```

...this will make the server connection finally work in the Android TV client via `http://jellyfin.service:8096`.

### Application version

0.17.6

### Where did you install the app from?

Sideloaded APK

### Device information

VAVA 4K Laser TV (2022)

### Android version

Android 7.1

### Jellyfin server version

10.9.11

**Comment by solidsnake1298** [comment]
Is there a separate A record for the hostname?

**Comment by whilenot-dev** [comment]
I just installed Firefox on my Android Box, and there seems to be the same issues regarding CNAME records (not working) <-> A records (working). **So I assume it's rather an issue with Android 7.1 and not with the _Jellyfin_ Android TV client**, sorry!

Thanks anyway! Closing the ticket now.
